pymongo/mongo_client.py

```diff
@@ -536,9 +537,13 @@ def __init__(
             connect = opts.get('connect', True)
         keyword_opts['tz_aware'] = tz_aware
         keyword_opts['connect'] = connect
-        # Validate all keyword options.
-        keyword_opts = dict(common.validate(k, v)
-                            for k, v in keyword_opts.items())
+        # Handle deprecated options in kwarg list.
+        keyword_opts = common.handle_option_deprecations(keyword_opts)
+        # Validate all kwarg options.
+        keyword_opts = common.get_validated_options(keyword_opts)
+        # Change kwarg option names to those used internally.
+        keyword_opts = common.normalize_options(keyword_opts)
```

Comment:
It seems like there should be a single method that performs these 3 steps. Don't we do the same 3 step process for URI options?

Reply:
I think the methods need to be separate even though in a majority of cases we perform all 3 steps on the URI options. This  is because we use these 3 methods in different ways depending on whether we are parsing (a) a URI string, or (b) keyword arguments passed to the `MongoClient`constructor. Also, we don't always want to perform the normalization step thereby requiring `normalize_options` to be a separate method that can be called after a boolean flag is checked. In light of all this, we must keep these three methods separate to facilitate re-use of this code.